### PR TITLE
refactor: dedupe scroll sync & guard swipes

### PR DESCRIPTION
**Pull request by CyanAutomation**
### Task Contract
```json
{
  "inputs": ["src/helpers/carousel/scrollSync.js", "src/helpers/carousel/swipe.js"],
  "outputs": ["src/helpers/carousel/scrollSync.js", "src/helpers/carousel/swipe.js"],
  "success": [
    "npx prettier . --check",
    "npx eslint .",
    "npm run check:jsdoc",
    "npx vitest run",
    "npx playwright test",
    "npm run check:contrast"
  ],
  "errorMode": "ask_on_public_api_change"
}
```

### Files Changed
- `src/helpers/carousel/scrollSync.js`: extract `syncPageFromScroll` helper to remove duplicated page update logic.
- `src/helpers/carousel/swipe.js`: guard touch handlers against empty touch arrays to avoid crashes.

### Verification
- `npx prettier . --check`
- `npx eslint .`
- `npm run check:jsdoc` *(fails: 165 functions missing or incomplete JSDoc blocks)*
- `npx vitest run`
- `npx playwright test` *(fails: 2 failing specs)*
- `npm run check:contrast`

### Risks
Touch handling changes may reveal edge cases on less common devices; failing Playwright specs and JSDoc coverage require follow-up.

------
https://chatgpt.com/codex/tasks/task_e_68bc989610f8832686b656a9750903da

### CODE REVIEW

File: src/helpers/carousel/scrollSync.js [hunk 1]
```diff
@@ -16,28 +16,24 @@ import { getPageMetrics } from "./metrics.js";
...
-    if (pageWidth > 0) {
-      const page = Math.round(ctrl.container.scrollLeft / pageWidth);
-      ctrl.currentPage = Math.max(0, Math.min(page, pageCount - 1));
-      ctrl.update();
-    }
+    if (pageWidth <= 0) return;
+    const page = Math.round(ctrl.container.scrollLeft / pageWidth);
+    ctrl.currentPage = Math.max(0, Math.min(page, pageCount - 1));
+    ctrl.update();
+  };
```
[the remark this answers is not in the document]
Comment (CyanAutomation) [reply src/helpers/carousel/scrollSync.js:28]: @codex fix comments
